packages/safe-apps-provider/src/provider.ts

```diff
@@ -7,44 +9,44 @@ const NETWORK_CHAIN_ID: Record<string, number> = {
   XDAI: 100,
 };
 
-// taken from ethers.js, compatible interface with web3 provider
-type AsyncSendable = {
-  isMetaMask?: boolean;
-  host?: string;
-  path?: string;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  sendAsync?: (request: any, callback: (error: any, response: any) => void) => void;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  send?: (request: any, callback: (error: any, response: any) => void) => void;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  request?: (request: { method: string; params?: Array<any> }) => Promise<any>;
-};
-
-export class SafeAppProvider implements AsyncSendable {
+// The API is based on Ethereum JavaScript API Provider Standard. Link: https://eips.ethereum.org/EIPS/eip-1193
+export class SafeAppProvider implements EIP1193Provider {
   private readonly safe: SafeInfo;
   private readonly sdk: SafeAppsSDK;
   private submittedTxs = new Map<string, Web3TransactionObject>();
+  private events = new EventEmitter();
 
   constructor(safe: SafeInfo, sdk: SafeAppsSDK) {
     this.safe = safe;
     this.sdk = sdk;
   }
 
-  public get chainId(): number {
-    return NETWORK_CHAIN_ID[this.safe.network];
+  async connect(): Promise<void> {
```

Comment:
If the implementation of this method returns `null`, is it following the specification regarding Connectivity?
https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1193.md#connectivity

Reply:
It doesn't really specify what should be returned. Also, it doesn't return null, it returns nothing (that defaults to undefined). I can see the only possible things are the connect/disconnect events, but I don't think they make much sense for the safe provider.

Reply:
Also shouldn't we emit the `ProviderConnectInfo` at least once?